app/proxy/client_test.go

```diff
@@ -108,29 +72,35 @@ func TestClientCallDoesNotReloadWalletIfAlreadyLoaded(t *testing.T) {
 	rand.Seed(time.Now().UnixNano())
 	walletID := sdkrouter.WalletID(rand.Intn(100))
 
-	mc := NewMockRPCClient()
-	c := &Client{rpcClient: mc}
-	q, _ := NewQuery(newRawRequest(t, "wallet_balance", nil))
+	srv := test.MockHTTPServer(nil)
+	defer srv.Close()
+
+	c := NewCaller(srv.URL, "")
+	q, err := NewQuery(newRawRequest(t, "wallet_balance", nil))
+	require.NoError(t, err)
 	q.SetWalletID(walletID)
 
-	mc.AddNextResponse(&jsonrpc.RPCResponse{
-		JSONRPC: "2.0",
-		Error: &jsonrpc.RPCError{
-			Message: "Couldn't find wallet: //",
-		},
-	})
-	mc.AddNextResponse(&jsonrpc.RPCResponse{
-		JSONRPC: "2.0",
-		Error: &jsonrpc.RPCError{
-			Message: "Wallet at path // is already loaded",
-		},
-	})
-	mc.AddNextResponse(&jsonrpc.RPCResponse{
-		JSONRPC: "2.0",
-		Result:  `"99999.00"`,
-	})
-
-	r, err := c.Call(q)
+	go func() {
+		srv.NextResponse <- test.ResToStr(t, jsonrpc.RPCResponse{
+			JSONRPC: "2.0",
+			Error: &jsonrpc.RPCError{
+				Message: "Couldn't find wallet: //",
+			},
+		})
+		srv.NextResponse <- "" // for the wallet_add call
```

Comment:
perhaps a matter of taste, but this would look more clean / self-explanatory:
EMPTY := ""

srv.NextResponse <- EMPTY

Reply:
i thought about that but didn't like that way. is `srv.NextResponse <- "{}"` better? or maybe an `SkipNext()` function on srv? i like SkipNext() less .

Reply:
I think SkipNext is more descriptive. Empty anything doesn't convey the intention and looks like a typo.